crates/hyperqueue/src/worker/parser.rs

```diff
@@ -208,12 +206,61 @@ pub fn parse_resource_definition(input: &str) -> anyhow::Result<ResourceDescript
     all_consuming(parse_resource_definition_inner()).parse_text(input)
 }
 
-pub fn parse_resource_coupling(input: &str) -> anyhow::Result<ResourceDescriptorCoupling> {
-    let names = input.split(",").map(|s| s.trim().to_string()).collect_vec();
-    if names.len() < 2 {
-        bail!("Resource coupling needs at least two resource names separated by commas");
-    }
-    Ok(ResourceDescriptorCoupling { names })
+fn parse_resource_coupling_group(
+    input: &str,
+    resources: &[ResourceDescriptorItem],
+) -> anyhow::Result<(u8, ResourceGroupIdx)> {
+    let input = input.trim();
```

Comment:
You gave up on the parser combinators? 😆 It's easier to have a high quality error message when parsing "manually", that's for sure...

Reply:
I am not giving up completely, but if I remember correctly, in this case, it was quite ugly code and manual parser was easy.